src/apps/vpn/platforms/linux/daemon/dbusservice.cpp

```diff
@@ -258,8 +291,126 @@ bool DBusService::firewallPid(int rootpid, const QString& state) {
 
 /* Clear the firewall and return all applications to the active state */
 bool DBusService::firewallClear() {
+  if (!checkCallerAuthz()) {
+    logger.error() << "Insufficient caller permissions";
+    return false;
+  }
+
   logger.debug() << "Clearing excluded app list";
   m_wgutils->resetAllCgroups();
   m_excludedApps.clear();
   return true;
 }
+
+/* Drop root permissions from the daemon. */
+void DBusService::dropRootPermissions() {
+  logger.debug() << "Dropping root permissions";
+
+  cap_t caps = cap_get_proc();
+  if (caps == nullptr) {
+    logger.warning() << "Failed to retrieve process capabilities";
+    return;
+  }
+  auto guard = qScopeGuard([&] { cap_free(caps); });
```

Comment:
Is this being used somewhere?

Reply:
This is just used for garbage collection. The scopeguard ensures that `cap_free()` is invoked regardless of how this function exits.